clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -591,12 +593,12 @@ private void failBatch(ProducerBatch batch, ProduceResponse.PartitionResponse re
 
     private void failBatch(ProducerBatch batch, long baseOffset, long logAppendTime, RuntimeException exception, boolean adjustSequenceNumbers) {
         if (transactionManager != null) {
-            if (exception instanceof OutOfOrderSequenceException
+            if ((exception instanceof OutOfOrderSequenceException || exception instanceof UnknownProducerIdException)
```

Comment:
Is the || needed ?
UnknownProducerIdException is subclass of OutOfOrderSequenceException